### PR TITLE
releases: add release notes for v0.8.0

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

File: releases/v0.8.0.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- IBM Secure Execution (SE) on IBM zSystems (s390x) running LinuxONE
+- AMD SEV-SNP
+
+The following platforms are in development:
+
+- ARM CCA (TODO: Find out what is the status of this?)
+
+## Limitations
+
+- TODO
```
Comment (fitzthum) [inline releases/v0.8.0.md:36]: You should copy the limitations from the previous release and then we will see if there is anything we can remove.

File: releases/v0.8.0.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## CVE Fixes
```
Comment (fitzthum) [inline releases/v0.8.0.md:38]: You can say `None` for this section.

File: releases/v0.8.0.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Release Notes for v0.8.0
+
+Release Date: Sep TBD, 2023
```
Comment (fitzthum) [inline releases/v0.8.0.md:3]: 9/27 wait did just jinx it

File: releases/v0.8.0.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+Please see the [quickstart guide](../quickstart.md) for details on how to try out Confidential
+Containers.
+
+Please refer to our [Acronyms](https://github.com/confidential-containers/documentation/wiki/Acronyms)
+and [Glossary](https://github.com/confidential-containers/documentation/wiki/Glossary) pages for a
+definition of the acronyms used in this document.
+
+## What's new
+
+- TODO
```
Comment (fitzthum) [inline releases/v0.8.0.md:14]: * Confidential Data Hub introduced on most platforms 
  * Supports sealed Kubernetes secrets that can be decrypted by KMS or stored in KBS
* Support added for multi-KBS model, where one KBS validates evidence using Attestation Service while another provides resource. 
* Something about policy @danmihai1
* Something host image pulling 
* Something about removing the containerd fork on most platforms

Comment (fidencio) [reply releases/v0.8.0.md:14]: > Something about policy @danmihai1

This is only `main` work.

Comment (fitzthum) [reply releases/v0.8.0.md:14]: Oh yeah

Comment (ChengyuZhu6) [reply releases/v0.8.0.md:14]: host image pulling with integrity(option) on most platforms(except s390x)
- support to download an image once on the host and sharing it as a block device with multiple pods or containers.
- not support downloading and sharing encrypted images on the host for the security of decryption keys.

Comment (ChengyuZhu6) [reply releases/v0.8.0.md:14]: guest pulling image without forked containerd on most platforms:
-  support image pulling in VM without forked containerd, while utilizing a remote-snapshotter on the host without the connection to VM.
- support one remote snapshotter: nydus-snapshotter.

File: releases/v0.8.0.md (hunk 1 above)
Comment (dcmiddle) [inline releases/v0.8.0.md:36]: Copied from v0.7 with minor update on the security badge status.

```suggestion
The following are known limitations of this release:

- Platform support is rapidly changing
  * Image signature validation with AMD SEV-ES is not covered by CI.
- SELinux is not supported on the host and must be set to permissive if in use.
- The generic KBS does not yet supported all platforms. 
- The format of encrypted container images is still subject to change
  * The [oci-crypt](https://github.com/containers/ocicrypt) container image format itself may still change
  * The tools to generate images are not in their final form
  * The image format itself is subject to change in upcoming releases
  * Not all image repositories support encrypted container images. 
- CoCo currently requires a custom build of `containerd`, which is installed by the operator.
  * Codepath for pulling images will change significantly in future releases.
  * `crio` is only supported with `cloud-api-adaptor`.
- Complete integration with Kubernetes is still in progress. 
  * OpenShift support is not yet complete.
  * Existing APIs do not fully support the CoCo security and threat model. [More info](https://github.com/confidential-containers/community/issues/53)
  * Some commands accessing confidential data, such as `kubectl exec`, may either fail to work, or incorrectly expose information to the host
  * Container images must be downloaded separately (inside guest) for each pod. [More info](https://github.com/confidential-containers/community/issues/66)
- The CoCo community aspires to adopting open source security best practices, but not all practices are adopted yet.
  * We track our status with the OpenSSF Best Practices Badge, which improved to 69% at the time of this release.
  * Vulnerability reporting mechanisms still need to be created. Public github issues are still appropriate for this release until private reporting is established.
```

**Review by fitzthum — COMMENTED** [review]
Two more notes

File: releases/v0.8.0.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+* Platform support is rapidly changing
+  * Image signature validation with AMD SEV-ES is not covered by CI.
+* SELinux is not supported on the host and must be set to permissive if in use.
+* The generic KBS does not yet supported all platforms.
+* The format of encrypted container images is still subject to change
+  * The [oci-crypt](https://github.com/containers/ocicrypt) container image format itself may still change
+  * The tools to generate images are not in their final form
+  * The image format itself is subject to change in upcoming releases
+  * Not all image repositories support encrypted container images.
+* CoCo currently requires a custom build of `containerd`, which is installed by the operator.
```
[the remark this answers is not in the document]
Comment (surajssd) [reply releases/v0.8.0.md:52]: Should I also remove the subsequent points after the `containerd` point. It seems like they are not exactly related to forked containerd? But the indentation adds to confusion, somehow making them sub-topics of this point.

[the remark this answers is not in the document]
Comment (fitzthum) [reply releases/v0.8.0.md:52]: You can just un-indent the two sub-points.

[the remark this answers is not in the document]
Comment (surajssd) [reply releases/v0.8.0.md]: done.

File: releases/v0.8.0.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+## What's new
+
+* Confidential Data Hub (CDH) allows users to do in-TEE secret processing.
+  * Supports sealed Kubernetes secrets that can be decrypted by KMS or stored in KBS
+* Support added for multi-KBS model, where one KBS validates evidence using Attestation Service while another provides resource.
+* Enable pulling container images on host which enables image sharing between multiple pods.
+* Upstream containerd now shipped in this release, with exceptions of the following projects:
+  * Enclave-cc
+  * TBD
```
Comment (fitzthum) [inline releases/v0.8.0.md:20]: Can remove the TBD

**Review by fitzthum — APPROVED** [review]
One minor note. Maybe we should try to merge this. I make another PR next week to update the date or anything else we find.

**Comment by surajssd** [comment]
> One minor note. Maybe we should try to merge this. I make another PR next week to update the date or anything else we find. 
> 
> 
> 
> @confidential-containers/confidential-containers-maintainers 

Or we can close this one and take the changes here for the new PR to reduce clutter.

File: releases/v0.8.0.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+* Intel SGX
+
+The following platforms are untested or partially supported:
+
+* IBM Secure Execution (SE) on IBM zSystems (s390x) running LinuxONE
+* AMD SEV-SNP
+
+The following platforms are in development:
+
+* ARM CCA (TODO: Find out what is the status of this?)
```
Comment (dcmiddle) [inline releases/v0.8.0.md:37]: Remove TODO. If there is sentiment that the CCA work is no longer active then we can drop the secion.
```suggestion
* ARM CCA
```

File: releases/v0.8.0.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
+# Release Notes for v0.8.0
+
+Release Date: Oct TBD, 2023
```
Comment (dcmiddle) [inline releases/v0.8.0.md:3]: Update date

**Review by dcmiddle — CHANGES_REQUESTED** [review]

File: releases/v0.8.0.md (hunk 7 above)
Comment (fitzthum) [reply releases/v0.8.0.md:37]: There is still ARM development happening, although idk what the timeilne is.

File: releases/v0.8.0.md (hunk 8 above)
Comment (fitzthum) [reply releases/v0.8.0.md:3]: We don't know the release date yet. I am thinking we can update in a subsequent PR next week.

Comment (fitzthum) [reply releases/v0.8.0.md:3]: I guess it definitely won't be in October tho

**Comment by dcmiddle** [comment]
> > One minor note. Maybe we should try to merge this. I make another PR next week to update the date or anything else we find.
> > @confidential-containers/confidential-containers-maintainers
> 
> Or we can close this one and take the changes here for the new PR to reduce clutter.

I'd prefer that we either merge this one and apply any last minute changes next week or keep it open until it is the final draft. 
I would not recommend closing this and opening a new one.

File: releases/v0.8.0.md (hunk 8 above)
Comment (dcmiddle) [reply releases/v0.8.0.md:3]: Yeah, sorry, I saw your note about updating the date after I posted this.

**Comment by wainersm** [comment]
@surajssd @fitzthum I'm talking with @stevenhorsman about the best way to flush peer-pods' news on CoCo's RN. So could you keep this open a few days more?

File: releases/v0.8.0.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+Please refer to our [Acronyms](https://github.com/confidential-containers/documentation/wiki/Acronyms)
+and [Glossary](https://github.com/confidential-containers/documentation/wiki/Glossary) pages for a
+definition of the acronyms used in this document.
+
+## What's new
+
+* Confidential Data Hub (CDH) allows users to do in-TEE secret processing.
+  * Supports sealed Kubernetes secrets that can be decrypted by KMS or stored in KBS
+* Support added for multi-KBS model, where one KBS validates evidence using Attestation Service while another provides resource.
+* Enable pulling container images on host which enables image sharing between multiple pods.
```
Comment (stevenhorsman) [inline releases/v0.8.0.md:17]: I'm not sure if this worked properly with the current version? See https://github.com/kata-containers/kata-containers/issues/8083

Comment (fitzthum) [reply releases/v0.8.0.md:17]: Yeah don't think so.

**Comment by fitzthum** [comment]
@wainersm yeah, I will probably take over this PR shortly and then we can merge it as the very last step.

**Comment by fitzthum** [comment]
Ok, I have commandeered this PR (thanks to @surajssd for getting it started). @dcmiddle PTAL.

@wainersm and @stevenhorsman lmk if there is anything to add for peer pods. 

I am thinking we should reworking the template for release notes a little bit for future releases, particularly the part about platform support. Currently it doesn't cover peer pods at all. We should highlight that we have cool stuff working with different CSPs.

File: releases/v0.8.0.md (hunk 6 above)
Comment (dcmiddle) [reply releases/v0.8.0.md:20]: I still see the TBD

File: releases/v0.8.0.md (hunk 7 above)
Comment (dcmiddle) [reply releases/v0.8.0.md:37]: Still see this TODO

File: releases/v0.8.0.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  * Container images must be downloaded separately (inside guest) for each pod. [More info](https://github.com/confidential-containers/community/issues/66)
+* The CoCo community aspires to adopting open source security best practices, but not all practices are adopted yet.
+  * We track our status with the OpenSSF Best Practices Badge, which improved to 69% at the time of this release.
+  * Vulnerability reporting mechanisms still need to be created. Public github issues are still appropriate for this release until private reporting is established.
+* Container metadata such as environment variables are not measured.
+* Kata Agent does not validate mount requests. A malicious host might be able to mount a shared filesystem into the PodVM.
+
+## CVE Fixes
+
+None
```
Comment (dcmiddle) [inline releases/v0.8.0.md:67]: add terminal end line

**Review by dcmiddle — APPROVED** [review]

**Comment by mythi** [comment]
> @wainersm and @stevenhorsman lmk if there is anything to add for peer pods.

@ChengyuZhu6 had valid feedback that is not included either:

> * Upstream containerd now shipped in this release,

so instead of this, we could be more verbose about the image pulling inside the guest using upstream containerd and nydys-snapshotter.

**Comment by wainersm** [comment]
> Ok, I have commandeered this PR (thanks to @surajssd for getting it started). @dcmiddle PTAL.
> 
> @wainersm and @stevenhorsman lmk if there is anything to add for peer pods.
>

@fitzthum looking at https://github.com/confidential-containers/cloud-api-adaptor/releases/tag/v0.8.0-alpha.1 I would list:

* Added support for pull image via remote-snapshotter (nydus-snapshotter implementation) 
* Improved peerpod-ctrl:
   * to check and clean old PeerPod objects
   * added libvirt ssh authentication support
   * Added support for ibmcloud
* Add CDH and ASR support
* Handle container registry authentication via userdata
* Added AMD SEV as well as IBM s390x support for the Libvirt provider
* Enabled public IP usage for pod VM on AWS and PowerVS providers
* webhook: added IBM ppc64le platform support
* Support adding custom tags to podvm instances
* Switched to launching CVM by default on AWS and Azure providers
* Added rollingUpdate strategy in cloud-api-adaptor daemonset

@stevenhorsman @bpradipt @surajssd ^ is this list accurate and complete?

> I am thinking we should rework the template for release notes a little bit for future releases, particularly the part about platform support. Currently it doesn't cover peer pods at all. We should highlight that we have cool stuff working with different CSPs.

**Comment by wainersm** [comment]
> > Ok, I have commandeered this PR (thanks to @surajssd for getting it started). @dcmiddle PTAL.
> > @wainersm and @stevenhorsman lmk if there is anything to add for peer pods.
> 
> @fitzthum looking at https://github.com/confidential-containers/cloud-api-adaptor/releases/tag/v0.8.0-alpha.1 I would list:
> 
>     * Added support for pull image via remote-snapshotter (nydus-snapshotter implementation)
> 
>     * Improved peerpod-ctrl:
>       
>       * to check and clean old PeerPod objects
>       * added libvirt ssh authentication support
>       * Added support for ibmcloud
> 
>     * Add CDH and ASR support
> 
>     * Handle container registry authentication via userdata
> 
>     * Added AMD SEV as well as IBM s390x support for the Libvirt provider
> 
>     * Enabled public IP usage for pod VM on AWS and PowerVS providers
> 
>     * webhook: added IBM ppc64le platform support
> 
>     * Support adding custom tags to podvm instances
> 
>     * Switched to launching CVM by default on AWS and Azure providers
> 
>     * Added rollingUpdate strategy in cloud-api-adaptor daemonset
> 
> 
> @stevenhorsman @bpradipt @surajssd ^ is this list accurate and complete?
> 

On the peer pods community meeting of this morning we came up with an updated list, @fitzthum :

   * Enabled image pulling via remote-snapshotter (nydus-snapshotter implementation) as an **experimental** feature. The `io.containerd.cri.runtime-handler: kata-remote` annotation should be set on the pod configuration to enable that feature.
   *  Improved peerpod-ctrl:
      * to check and clean old PeerPod objects
      * added libvirt ssh authentication support
       * Added support for ibmcloud
   * Added runtime secure key release for image decryption via remote attestation on AKS
   * Handle container registry authentication via userdata
   * Added AMD SEV as well as IBM s390x support for the Libvirt provider
   * Enabled public IP usage for pod VM on AWS and PowerVS providers
   * webhook: added IBM ppc64le platform support
   * Support adding custom tags to podvm instances
   * Switched to launching CVM by default on AWS and Azure providers
   * Added rollingUpdate strategy in cloud-api-adaptor daemonset
   * Disabled secureboot by default

The following applies to local hypervisor too so you might want to put in a separated section:

   *  imagePullSecrets is now required to pull images from authenticated registry
  
@stevenhorsman @katexochen @mkulke ^ let me know if did not capture the list correctly :)

**Comment by mkulke** [comment]
@wainersm 

> Added runtime secure key release for image decryption via remote attestation on AKS

Support for secure key release at runtime and image decryption via remote attestation on AKS.

File: releases/v0.8.0.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+  * In this release images are still pulled inside the guest.
+  * **Nydus snapshotter requires the following annotation for each pod `io.containerd.cri.runtime-handler: <runtime-class`.**
+  * Nydus snapshotter support is still evolving. See limitations section below for details.
+* A new component, the Confidential Data Hub (CDH) is now deployed inside the guest.
+  * CDH is an evolution of the Attestation Agent that supports advanced features.
+  * CDH supports sealed Kubernetes secrets which are managed by the control plane, but securely unwrapped inside the enclave.
+  * CDH supports connections to both KBS and KMS.
+* New architecture of Attestation Agent and CDH allows a client to deploy multiple KBSes.
+  * One KBS can be used for validating evidence with the Attestation Service while another can provide resources.
+* Pulling from an authenticated registry now uses `imagePullSecrets`.
```
Comment (stevenhorsman) [inline releases/v0.8.0.md:25]: I think it's more that
Pulling from an authenticated registry now *requires* `imagePullSecrets` even when pulling on the guest as the nydus-snapshotter need the secret to pull down the metadata. Previously the creds could just be passed in via the attestation-agent's getresources.

Sorry if I wasn't clear in explaining it (via Wainer's update).

Comment (fitzthum) [reply releases/v0.8.0.md:25]: Ok, make sense.

Comment (fitzthum) [reply releases/v0.8.0.md:25]: How does that relate to `Handle container registry authentication via userdata` ?

Comment (stevenhorsman) [reply releases/v0.8.0.md:25]: Good question. @wainersm can confirm, but I think that line is referring to how we allow the user to pass through the credentials into the peer pod (to be used in the guest pull by image-rs) via userdata/metadata-serviec, which means that they can use a common image without having to custom build it. The technical mechanism of doing it has updated, but the user experience hasn't too much here compared to previous releases, we are now using the agent-config option that Ding added, rather going via the offline_fs_kbc-resource.kbon.

Pre-nydus-snapshotter this was enough to pull authenticated registry images, but as with local hypervisor we now need the imagePullSecrets as well. I hope that helps a bit?

**Comment by fitzthum** [comment]
I added two more commits to fixup some other things in the docs, such as adding the runtime handler annotation to all the yaml examples.

File: releases/v0.8.0.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+* Intel SGX
+
+The following platforms are untested or partially supported:
+
+* IBM Secure Execution (SE) on IBM zSystems (s390x) running LinuxONE
+* AMD SEV-SNP
+
+The following platforms are in development:
+
+* ARM CCA
```
Comment (chendave) [inline releases/v0.8.0.md:57]: ARM CCA has been supported I think, pls refer to:

- https://github.com/confidential-containers/attestation-service/pull/90
- https://github.com/confidential-containers/guest-components/pull/321
- https://github.com/confidential-containers/kbs/pull/76

A native verifier is on the way, but this is a different story.

Comment (chendave) [reply releases/v0.8.0.md:55]: thanks Tobin!

**Pull request merged**